src/back/TicketManager/src/Core/TicketManager.Core.Services/Endpoints/Organizers/OrganizerDecideValidator.cs

```diff
@@ -0,0 +1,46 @@
+using FastEndpoints;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using TicketManager.Core.Contracts.Organizers;
+using TicketManager.Core.Domain.Organizer;
+using TicketManager.Core.Services.Services.Mockables;
+using TicketManager.Core.Services.ValidationExtensions;
+
+namespace TicketManager.Core.Services.Endpoints.Organizers;
+
+public class OrganizerDecideValidator : Validator<OrganizerDecideRequest>
+{
+    private IServiceScopeFactory scopeFactory;
+    private MockableCoreDbResolver dbResolver;
+    
+    public OrganizerDecideValidator(IServiceScopeFactory scopeFactory, MockableCoreDbResolver dbResolver)
+    {
+        this.scopeFactory = scopeFactory;
+        this.dbResolver = dbResolver;
+
+        RuleFor(req => req.OrganizerId)
+            .MustAsync(IsIdPresent)
+            .WithCode(OrganizerDecideRequest.ErrorCodes.OrganizerNotInDatabase)
+            .MustAsync(IsUnverified)
+            .WithCode(OrganizerDecideRequest.ErrorCodes.OrganizerAlreadyVerified);
+    }
+    
+    private async Task<bool> IsIdPresent(Guid id, CancellationToken cancellationToken)
+    {
+        using var scope = scopeFactory.CreateScope();
+        
+        return await dbResolver.Resolve(scope)
+            .Organizers
+            .AnyAsync(o => o.Id == id, cancellationToken);
+    }
+    
+    private async Task<bool> IsUnverified(Guid id, CancellationToken cancellationToken)
+    {
+        using var scope = scopeFactory.CreateScope();
+        
+        return await dbResolver.Resolve(scope)
+            .Organizers
+            .AnyAsync(o => o.Id == id && o.VerificationStatus == VerificationStatus.Unverified, cancellationToken);
+    }
```

Comment:
With some tricks with extensions methods it would be possible to do only one db call here. But let's say it's task for the future.